src/uu/chmod/src/chmod.rs

```diff
@@ -37,13 +37,54 @@ mod options {
 
 #[uucore::main]
 pub fn uumain(args: impl uucore::Args) -> UResult<()> {
-    let mut args = args.collect_lossy();
+    let args = args.collect_lossy();
 
-    // Before we can parse 'args' with clap (and previously getopts),
-    // a possible MODE prefix '-' needs to be removed (e.g. "chmod -x FILE").
-    let mode_had_minus_prefix = mode::strip_minus_from_mode(&mut args);
+    // "-w -w" is the same as "-w,-w"
+    // "-w file -w" is the same as "-w,-w file"
+    // "-- -w file" is the same as "-w file"
+    let mut parsed_cmode_vec = Vec::new();
+    let double_hyphen = args.iter().find(|&x| x == "--");
+    let mut args_iter = args.iter();
+    let mut clean_args: Vec<&String> = args_iter
+        .by_ref()
+        .skip(1)
+        .take_while(|&a| a != "--")
+        .filter(|a| {
+            if a.len() >= 2
+                && a.starts_with('-')
+                && matches!(
+                    a.chars().nth(1).unwrap(),
+                    'r' | 'w' | 'x' | 'X' | 's' | 't' | 'u' | 'g' | 'o' | '0'..='7'
+                )
+            {
+                parsed_cmode_vec.push(a.to_string());
+                false
+            } else {
+                true
+            }
+        })
+        .collect();
+    if let Some(double_hyphen) = double_hyphen {
+        clean_args.push(double_hyphen);
+    }
+    let parsed_cmode = if parsed_cmode_vec.is_empty() {
+        args_iter.next().cloned()
+    } else {
+        Some(parsed_cmode_vec.join(","))
+    };
+    clean_args.extend(args_iter);
 
-    let matches = uu_app().after_help(LONG_USAGE).try_get_matches_from(args)?;
+    let mut args_vec = vec![&args[0]];
+    let pseudo_cmode = "w".to_string();
+    if parsed_cmode.is_some() {
+        // pseudo cmode is needed to satisfy clap, but won't be used later
```

Comment:
Maybe we should remove the mode from clap? It will still show up in the usage so I think we don't need it to be a clap argument

Reply:
Currently, the "normal mode usage" (`chmod <mode> <file>`, mode not starting with `-`) is still handled by clap. I guess to be able to do that, we have to add logic for this case too. I guess the same is valid for non-utf8 inputs.

Reply:
Alright the pseudo mode is fine then